docs/en/upload.md

```diff
@@ -62,11 +62,12 @@ print("Connecting to WiFi access point...")
 wifi.setmode(wifi.STATION)
 wifi.sta.config(SSID, PASSWORD)
 -- wifi.sta.connect() not necessary because config() uses auto-connect=true by default
-tmr.alarm(1, 1000, 1, function()
+wifitimer = tmr.create()
+wifitimer:alarm(1000, tmr.ALARM_AUTO, function()
     if wifi.sta.getip() == nil then
         print("Waiting for IP address...")
     else
-        tmr.stop(1)
+        wifitimer:stop()
```

Comment:
Great input Philip! The callback _does_ get timer object back, I just tested:

```lua
mytimer = tmr.create()
mytimer:alarm(1000, tmr.ALARM_AUTO, function(cb_tmr) 
    cb_tmr:stop()
    print("hey there")
    print(cb_tmr:state())
end)
```
Prints
> hey there
false	1

and stops the timer. I'll document the missing `func` parameter and update this example.